src/kp/kp_client.py

```diff
@@ -120,14 +129,33 @@ def request_session_token(self):
             if cookie['name'] == 'JSESSIONID':
                 session_cookie = cookie['value']
                 break
+
         self.session_cookie = session_cookie
-        return session_cookie
+        self.isFetchingToken = False
+        return self.session_cookie
 
     def authenticate(self):
+        while self.isFetchingToken:
```

Comment:
How about a while with timeout ?

```python
timeout = time.time() + 60*3   # 3 minutes from now
while self.isFetchingToken:
    if time.time() > timeout:
        break
```

Reply:
I added timeout to the fetch itself. Function request_session_token should never get stuck.

https://github.com/Randers-Kommune-Digitalisering/api-service/blob/29a0a2004d1f275f4be92fd4f8b6ac0160689993/src/kp/kp_client.py#L113C12-L114C35

Reply:
maybe both to be sure? This would have negated the issue with `self.isFetchingToken = False` not being set on an exception.

Reply:
I added both.